datastore/mongodb/filter_outputs.go

```diff
@@ -50,30 +50,52 @@ func (c *Client) CreateFilterOutput(ctx context.Context, f *model.FilterOutput)
 func (c *Client) UpdateFilterOutput(ctx context.Context, f *model.FilterOutput) error {
 	col := c.collections.filterOutputs
```

Comment:
Realise you're just editing an existing method, but wonder if it's worth locking the table here.